Description: revert https://github.com/joyent/node/commit/f2a45caf
 Unfortunately it relies on a libv8 patch that breaks v8 3.14.5 API.
 This is part 1/2.
Forwarded: not-needed
Last-Update: 2015-05-04
--- a/src/node.js
+++ b/src/node.js
@@ -216,14 +216,6 @@
     return startup._lazyConstants;
   };
 
-  function _clearDomainsStack() {
-    var domainModule = NativeModule.require('domain');
-    var domainStack = domainModule._stack;
-    domainStack.length = 0;
-    domainModule.active = null;
-    process.domain = null;
-  }
-
   startup.processFatal = function() {
     // call into the active domain, or emit uncaughtException,
     // and exit if there are no listeners.
@@ -276,6 +268,13 @@
             // If caught is false after this, then there's no need to exit()
             // the domain, because we're going to crash the process anyway.
             caught = domain.emit('error', er);
+
+            // Exit all domains on the stack.  Uncaught exceptions end the
+            // current tick and no domains should be left on the stack
+            // between ticks.
+            var domainModule = NativeModule.require('domain');
+            domainStack.length = 0;
+            domainModule.active = process.domain = null;
           } catch (er2) {
             // The domain error handler threw!  oh no!
             // See if another domain can catch THIS error,
@@ -292,11 +291,6 @@
             }
           }
         }
-
-        // Exit all domains on the stack.  Uncaught exceptions end the
-        // current tick and no domains should be left on the stack
-        // between ticks.
-        _clearDomainsStack();
       } else {
         caught = process.emit('uncaughtException', er);
       }
--- a/test/simple/test-domain-top-level-error-handler-clears-stack.js
+++ /dev/null
@@ -1,41 +0,0 @@
-// Copyright Joyent, Inc. and other Node contributors.
-//
-// Permission is hereby granted, free of charge, to any person obtaining a
-// copy of this software and associated documentation files (the
-// "Software"), to deal in the Software without restriction, including
-// without limitation the rights to use, copy, modify, merge, publish,
-// distribute, sublicense, and/or sell copies of the Software, and to permit
-// persons to whom the Software is furnished to do so, subject to the
-// following conditions:
-//
-// The above copyright notice and this permission notice shall be included
-// in all copies or substantial portions of the Software.
-//
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
-// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
-// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
-// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
-// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
-// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
-// USE OR OTHER DEALINGS IN THE SOFTWARE.
-
-var assert = require('assert');
-var domain = require('domain');
-
-/*
- * Make sure that the domains stack is cleared after a top-level domain
- * error handler exited gracefully.
- */
-var d = domain.create();
-
-d.on('error', function() {
-  process.nextTick(function() {
-    if (domain._stack.length !== 1) {
-      process.exit(1);
-    }
-  });
-});
-
-d.run(function() {
-  throw new Error('Error from domain');
-});
